test-validator/src/lib.rs

```diff
@@ -203,6 +206,37 @@ impl TestValidatorGenesis {
         self
     }
 
+    pub fn add_accounts_with_path(&mut self, filenames: Vec<&str>) -> &mut Self {
```

Comment:
How about:
```suggestion
    pub fn add_accounts_from_json_files(&mut self, filename: &[Pubkey]) -> &mut Self {
```

`Pubkey` as the type for filename, and we don't need to assume a `Vec` when any slice will do

Reply:
It just feels like better typing to require the filename to be a `Pubkey` here to me, we don't actually accept any other value right?  So we can push the validation out to the `--account-from-json-file` clap.rs argument handling, and know that internally the filename will always be a valid pubkey

Reply:
Oh, I was thinking we'd just use the filename as the address:
```
$ solana-test-validator --account-from-json-file SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt.json
```
but yeah maybe that's not so great, and it doesn't match what `--bpf-program` already does.

So I take it all back, something like this seems great (also looking to drop the "clone" naming from this argument):
```
$ solana-test-validator --account SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt SRM_token.json
```

Reply:
Yes `account` as argument name sounds straightforward enough. As a final round on the argument structure, I propose two options:
1. I remove the pubkey from what gets serialized as part of `solana account --output-file ...` and we use what you just mentioned:
```sh
$ solana-test-validator --account SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt SRM_token.json
```
2. I leave the pubkey as part of the serialized data, and we simply point to the file (the address the account gets loaded to being the one inside the file)
```sh
$ solana-test-validator --account SRM_token.json
```

I mostly want to avoid confusion due to specifying the program address in 2 separate places, which could lead to confusion, and require either prioritizing one over the other, or enforcing that they both be equal.

Reply:
I like it!  

As a minor tweak, we could change the address to be an optional second argument to `--account` will simplify the arg parsing code. Slightly inconsistent with the `--bpf-program` argument though, but feels more natural if the address is also now embedded in SRM_token.json.

```
$ solana-test-validator --account SRM_token.json
        or
$ solana-test-validator --account SRM_token.json SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt 
```

Reply:
1. My bad, I was not clear enough. I was actually proposing to implement one, **or** the other. I just looked into implementing an optional pubkey arg like you suggest, and unless I am missing something it makes the parsing quite tricky, because option values are collected "flat", even across separate occurrences of a particular option.
Example: `solana-test-validator --account serum SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt uscd --account raydium` would be collected by `clap` as the following `String` array: `[serum, SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt, usdc, raydium]`.
The grouping would then need to be inferred from a `is_pubkey` check succeeding or not, or something similar.
Do you think this is worth it? Or would 1 of the options above sound good enough?

2. A detail I'd rather clarify, since I mislead you a bit in the examples above: the file argument is currently *without* the file extension, which is automatically appended to the argument provided. I went that way because only a json file is valid, but I'm open to challenge if you think that's weird.

Reply:
Got it.  This seems best then
```
$ solana-test-validator --account SRMuApVNdxXokk5GT7XD5cUUgXMBCoAz2LHeuAoKWRt SRM_token.json
```
Consistent with `--bpf-program` and allows the same `SRM_token.json` account data to be reused at different addresses without editing the file contents

Reply:
All comments addressed. CI under way. I actually changed the parsing so that the whole filename (including extension) is required. Didn't make much sense to not allow the user to add it (if they're using autocompletion for instance).

The only thing a bit annoying right now is that if you specify a file that does not exist in CWD or tests/fixtures (the paths explored by `solana_program_test::find_file`), the error log is in `test-ledger/validator.log`, and not in the CLI output like it is for a wrong pubkey.
I left it that way as it is a tradeoff to be able to get the files automatically found in `tests/fixtures` without having to supply the whole path. Not sure if it is worth it though, or if I should just require a straight valid path which could be validated before starting the validator, and would display the error in the command output.

Reply:
Looks great, thanks.  I pushed a tiny commit to update the arg name